src/helpers/utils.ts

```diff
@@ -103,23 +103,54 @@ export function runCommand(cmd: string, requiredRepoType = RepoType.Undefined) {
   } else if (getRepoType() !== requiredRepoType) {
     vscode.window.showErrorMessage(`Command works only with ${requiredRepoType} projects!`);
   } else {
-    const project = resolve(workspace.uri.fsPath, 'package.json');
-
-    try {
-      const { scripts = {} } = __non_webpack_require__(project) || {};
-      const candidates = Object.keys(scripts).filter((m) => scripts[m].trim().startsWith(cmd));
-      const shellCommand =
-        candidates.length === 0 ? cmd : candidates.length === 1 ? scripts[candidates.pop() ?? ''] : undefined;
-
-      if (shellCommand !== undefined) {
-        execCommand(shellCommand);
+    const piralOrPilet = cmd.split(' ')[1];
+    const nodeModulesAreAvailable = existsSync(`${workspace.uri.fsPath}/node_modules`);
+    const piralCliIsAvailable = existsSync(`${workspace.uri.fsPath}/node_modules/.bin/${piralOrPilet}`);
+
+    if (!nodeModulesAreAvailable) {
+      // ask user to install node-modules
+      vscode.window
+        .showInformationMessage(
+          'Dependencies are not installed yet, should we install the dependencies now?',
+          'Yes',
+          'No',
+        )
+        .then((answer) => {
+          if (answer === 'Yes') {
+            execCommand('npm i');
+          }
+        });
+    } else {
+      if (!piralCliIsAvailable) {
+        // ask user to install piral-cli
+        vscode.window
+          .showInformationMessage('Piral CLI not installed yet, should we install the dependencies now?', 'Yes', 'No')
+          .then((answer) => {
+            if (answer === 'Yes') {
+              execCommand('npm i piral-cli');
+            }
+          });
       } else {
-        vscode.window.showQuickPick(candidates).then(execCommand);
+        // execute the command - cmd
+        const project = resolve(workspace.uri.fsPath, 'package.json');
+
+        try {
+          const { scripts = {} } = __non_webpack_require__(project) || {};
+          const candidates = Object.keys(scripts).filter((m) => scripts[m].trim().startsWith(cmd));
+          const shellCommand =
+            candidates.length === 0 ? cmd : candidates.length === 1 ? scripts[candidates.pop() ?? ''] : undefined;
+
+          if (shellCommand !== undefined) {
+            execCommand(shellCommand);
+          } else {
+            vscode.window.showQuickPick(candidates).then(execCommand);
+          }
+        } catch (err) {
+          vscode.window.showErrorMessage(
```

Comment:
This should be the first check.

If we don't have a valid workspace / Piral project (which requires a package.json with a `piral-cli` in there as some dependency) then this should not trigger at all.